Spigot-Server-Patches/0212-Use-TerminalConsoleAppender-for-console-improvements.patch

```diff
@@ -0,0 +1,659 @@
+@@ -14,10 +14,11 @@ import java.util.logging.Logger;
+ import joptsimple.OptionParser;
+ import joptsimple.OptionSet;
+ import net.minecraft.server.MinecraftServer;
+-import org.fusesource.jansi.AnsiConsole;
++import net.minecrell.terminalconsole.TerminalConsoleAppender;
++//import org.fusesource.jansi.AnsiConsole; // Paper
+ 
+ public class Main {
+-    public static boolean useJline = true;
++    //public static boolean useJline = true; // Paper
+     public static boolean useConsole = true;
+ 
+     public static void main(String[] args) {
+@@ -170,6 +171,8 @@ public class Main {
+             }
+ 
+             try {
++                // Paper start - Handled by TerminalConsoleAppender
++                /*
+                 // This trick bypasses Maven Shade's clever rewriting of our getProperty call when using String literals
+                 String jline_UnsupportedTerminal = new String(new char[] {'j','l','i','n','e','.','U','n','s','u','p','p','o','r','t','e','d','T','e','r','m','i','n','a','l'});
+                 String jline_terminal = new String(new char[] {'j','l','i','n','e','.','t','e','r','m','i','n','a','l'});
+@@ -188,10 +191,16 @@ public class Main {
+                     // This ensures the terminal literal will always match the jline implementation
+                     System.setProperty(jline.TerminalFactory.JLINE_TERMINAL, jline.UnsupportedTerminal.class.getName());
+                 }
+-
++                */
++                if (options.has("nojline")) {
++                    // Disable JLine manually
++                    System.setProperty(TerminalConsoleAppender.JLINE_OVERRIDE_PROPERTY, "false");
++                }
++                // Paper end
+ 
+                 if (options.has("noconsole")) {
+                     useConsole = false;
++                    System.setProperty(TerminalConsoleAppender.JLINE_OVERRIDE_PROPERTY, "false"); // Paper
+                 }
+ 
+                 if (Main.class.getPackage().getImplementationVendor() != null && System.getProperty("IReallyKnowWhatIAmDoingISwear") == null) {
+diff --git a/src/main/java/org/bukkit/craftbukkit/command/ColouredConsoleSender.java b/src/main/java/org/bukkit/craftbukkit/command/ColouredConsoleSender.java
+deleted file mode 100644
+index 26a2fb89..00000000
+--- a/src/main/java/org/bukkit/craftbukkit/command/ColouredConsoleSender.java
++++ /dev/null
+@@ -1,74 +0,0 @@
+-package org.bukkit.craftbukkit.command;
+-
+-import java.util.EnumMap;
+-import java.util.Map;
+-
+-import org.fusesource.jansi.Ansi;
+-import org.fusesource.jansi.Ansi.Attribute;
+-import jline.Terminal;
+-
+-import org.bukkit.Bukkit;
+-import org.bukkit.ChatColor;
+-import org.bukkit.command.ConsoleCommandSender;
+-import org.bukkit.craftbukkit.CraftServer;
+-
+-public class ColouredConsoleSender extends CraftConsoleCommandSender {
+-    private final Terminal terminal;
+-    private final Map<ChatColor, String> replacements = new EnumMap<ChatColor, String>(ChatColor.class);
```

Comment:
Id say we should leave this in, so that if a plugin is dumb, it will fail more gracefully (so only the dumb part of the plugin dies)

Also, if upstream changes this file, it would cause conflicts for us.